services/QuillLMS/engines/evidence/lib/evidence/evidence/plagiarism_check.rb

```diff
@@ -125,12 +125,23 @@ def feedback_object
 
     private def identify_matched_slices(entry_slices, passage_slices)
       entry_slices.select do |_, slice|
+        # Check to see if there's enough similarity for it to be worth doing a Levenshtein comparison
+        next false unless confirm_minimum_overlap(slice, passage_slices)
         slice_string = slice.join(' ')
         passage_slice_strings = passage_slices.map { |s| s.join(' ') }
         passage_slice_strings.any? { |passage_string| DidYouMean::Levenshtein.distance(slice_string, passage_string) <= FUZZY_CHARACTER_THRESHOLD }
       end.keys
     end
 
+    private def confirm_minimum_overlap(target_array, source_arrays)
```

Comment:
take or leave, consider using Ruby boolean method naming conventions with question mark and readable verb, e.g.

```ruby
def has_minimum_overlap?(target_array, source_arrays)

Reply:
Good call. I actually prefer the convention for readability, so thanks for reminding me I'd missed it here.